pkg/apis/cluster/v1alpha1/types.go

```diff
@@ -14,6 +14,8 @@ const (
 	ResourcePluralCluster = "clusters"
 	// ResourceNamespaceScopedCluster indicates if Cluster is NamespaceScoped.
 	ResourceNamespaceScopedCluster = false
+	// ClusterNameAnnotation is annotation key of cluster name.
+	ClusterNameAnnotation = "cluster.karmada.io/name"
```

Comment:
We can move this key to the `well_known_annotations.go` file.

@RainbowMango How do you think this `ClusterNameAnnotation` name?

Reply:
That's exactly what I'm thinking about.
Wait for a while, let me see if there is a better name.

Reply:
😄，naming is harder than coding.

Reply:
@RainbowMango @XiShanYongYe-Chang  how about `ClusterAnnotationKeyName`?

Reply:
Better than the previous :)